src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -0,0 +1,180 @@
+import json
+from copy import deepcopy
+from typing import List, Dict, Any
+
+from pydantic import Field, BaseModel
+
+from marqo.core import constants as index_constants
+from marqo.core.exceptions import VespaDocumentParsingError
+from marqo.core.unstructured_vespa_index import common as unstructured_common
+from marqo.core.unstructured_vespa_index import constants as unstructured_constants
+
+
+class UnstructuredVespaDocumentFields(BaseModel):
+    """A class with fields that are common to all Vespa documents."""
+    marqo__id: str = Field(alias=unstructured_common.VESPA_FIELD_ID)
+
+    strings: List[str] = Field(default_factory=list, alias=unstructured_common.STRINGS)
+    long_string_fields: Dict[str, str] = Field(default_factory=dict, alias=unstructured_common.LONGS_STRINGS_FIELDS)
+    short_string_fields: Dict[str, str] = Field(default_factory=dict, alias=unstructured_common.SHORT_STRINGS_FIELDS)
+    string_arrays: List[str] = Field(default_factory=list, alias=unstructured_common.STRING_ARRAY)
+    int_fields: Dict[str, int] = Field(default_factory=dict, alias=unstructured_common.INT_FIELDS)
+    bool_fields: Dict[str, bool] = Field(default_factory=dict, alias=unstructured_common.BOOL_FIELDS)
+    float_fields: Dict[str, float] = Field(default_factory=dict, alias=unstructured_common.FLOAT_FIELDS)
+    score_modifiers_fields: Dict[str, Any] = Field(default_factory=dict, alias=unstructured_common.SCORE_MODIFIERS)
+    vespa_chunks: List[str] = Field(default_factory=list, alias=unstructured_common.VESPA_DOC_CHUNKS)
+    vespa_embeddings: Dict[str, Any] = Field(default_factory=dict, alias=unstructured_common.VESPA_DOC_EMBEDDINGS)
+    vespa_multimodal_params: Dict[str, str] = Field(default_factory=str,
+                                                    alias=unstructured_common.VESPA_DOC_MULTIMODAL_PARAMS)
+    vector_counts: int = Field(default=0, alias=unstructured_common.FIELD_VECTOR_COUNT)
+
+    match_features: Dict[str, Any] = Field(default_factory=dict, alias=unstructured_common.VESPA_DOC_MATCH_FEATURES)
+
+    # Fields that are excluded when generating vespa documents
+    _EXCLUDED_FIELDS = {"match_features"}
+
+    class Config:
+        allow_population_by_field_name = True
+
+    def to_vespa_dictionary(self) -> Dict[str, Any]:
+        # Exclude None and empty fields but keep 0 as we need to have vector_counts in docs
+        return {k: v for k, v in self.dict(exclude_none=True, by_alias=True,
+                                           exclude=self._EXCLUDED_FIELDS).items() if v or v == 0}
```

Comment:
Do you need this exclusion? You will only have `matchfeatures` in search response